test6/ResourceLocalizationService.java

```
@@ -0,0 +1,1714 @@
+  private LocalizerTracker localizerTracker;
+  private RecordFactory recordFactory;
+  private final ScheduledExecutorService cacheCleanup;
+  private LocalizerTokenSecretManager secretManager;
+  private NMStateStoreService stateStore;
+  @VisibleForTesting
+  final NodeManagerMetrics metrics;
+
+  @VisibleForTesting
+  LocalResourcesTracker publicRsrc;
+
+  private LocalDirsHandlerService dirsHandler;
+  private DirsChangeListener localDirsChangeListener;
+  private DirsChangeListener logDirsChangeListener;
+  private Context nmContext;
+  private DiskValidator diskValidator;
+
+  /**
+   * Map of LocalResourceTrackers keyed by username, for private
+   * resources.
+   */
+  @VisibleForTesting
+  final ConcurrentMap<String, LocalResourcesTracker> privateRsrc =
+    new ConcurrentHashMap<String,LocalResourcesTracker>();
+
+  /**
+   * Map of LocalResourceTrackers keyed by appid, for application
+   * resources.
+   */
+  private final ConcurrentMap<String,LocalResourcesTracker> appRsrc =
+    new ConcurrentHashMap<String,LocalResourcesTracker>();
+  
+  FileContext lfs;
+
+  public ResourceLocalizationService(Dispatcher dispatcher,
+      ContainerExecutor exec, DeletionService delService,
+      LocalDirsHandlerService dirsHandler, Context context,
+      NodeManagerMetrics metrics) {
+
+    super(ResourceLocalizationService.class.getName());
+    this.exec = exec;
+    this.dispatcher = dispatcher;
+    this.delService = delService;
+    this.dirsHandler = dirsHandler;
+
+    this.cacheCleanup = new HadoopScheduledThreadPoolExecutor(1,
+        new ThreadFactoryBuilder()
+          .setNameFormat("ResourceLocalizationService Cache Cleanup")
+          .build());
+    this.stateStore = context.getNMStateStore();
+    this.nmContext = context;
+    this.metrics = metrics;
+  }
+
+  FileContext getLocalFileContext(Configuration conf) {
+    try {
+      return FileContext.getLocalFSFileContext(conf);
+    } catch (IOException e) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)